staging/src/k8s.io/api/resource/v1alpha3/types.go

```diff
@@ -287,6 +295,22 @@ type DeviceAttribute struct {
 	VersionValue *string `json:"version,omitempty" protobuf:"bytes,5,opt,name=version"`
 }
 
+// NullableDeviceAttribute must have exactly one field set.
+// It has the exact same fields as a DeviceAttribute plus `null` as
+// an additional alternative.
+type NullableDeviceAttribute struct {
+	DeviceAttribute `json:",inline" protobuf:"bytes,1,opt,name=deviceAttribute"`
```

Comment:
We should put a note in DeviceAttribute that it can never add a `NullValue` field because this will break.